pkg/controllers/job/job_controller_actions.go

```diff
@@ -452,6 +468,78 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 	return nil
 }
 
+func (cc *jobcontroller) waitDependsOnTaskMeetCondition(taskName string, podToCreateEachTask []*v1.Pod, job *batch.Job) {
+	taskIndex := jobhelpers.GetTaskIndex(taskName, job)
+	if job.Spec.Tasks[taskIndex].DependsOn != nil {
+		dependsOn := *job.Spec.Tasks[taskIndex].DependsOn
+		if len(dependsOn.Name) > 1 && dependsOn.Iteration == batch.IterationAny {
+			wait.PollInfinite(100*time.Millisecond, func() (bool, error) {
+				for _, DependsOnTask := range dependsOn.Name {
+					if cc.isDenpendentPodsReady(DependsOnTask, job) {
+						return true, nil
+					}
+				}
+				return false, nil
+			})
+		} else {
+			for _, DependsOnTask := range dependsOn.Name {
+				wait.PollInfinite(100*time.Millisecond, func() (bool, error) {
+					if cc.isDenpendentPodsReady(DependsOnTask, job) {
+						return true, nil
+					}
+					return false, nil
+				})
+			}
+		}
+	}
+}
+
+func (cc *jobcontroller) isDenpendentPodsReady(DependsOnTask string, job *batch.Job) bool {
+	dependentPod := jobhelpers.GetPodsInTask(DependsOnTask, job)
```

Comment:
dependentPod => dependentPods
And it is better to unify using dependon or dependent, not mixed

Reply:
I agree with what you said, which one do you think is better?